src/DGtal/geometry/helpers/ScaleProfile.h

```diff
@@ -0,0 +1,406 @@
+
+
+ 
+    /**
+     * Clears the object as if it has been just created.
+     */
+    void clear();
+
+
+
+    /**
+     * Initializer. Must be called before adding datas. Specifies the
+     * scales of the profile (generally an iterator on a sequence
+     * (1,2,3,4,...N)).
+     *
+     * @param[in] beginScale an iterator pointing on the first scale (some
+     * floating-point convertible value).
+     * @param[in] endScale an iterator pointing after the last scale.
+     * @param[in] storeValsInStats flag to store values in statistics (so that 
+     *    the median value is accessible (default false)). 
+     */
+    template <typename Iterator>
+    void init(  Iterator beginScale,  Iterator endScale, 
+                const bool storeValsInStats=false );
+
+
+
+    /**
+     * Initializer. Must be called before adding datas. Specifies the
+     * scales of the profile as the sequence (1,2,3,4,...,nb).
+     *
+     * @param[in] nb an integer number strictly positive.
+     * @param[in] storeValsInStats flag to store values in statistics (so that 
+     *   the median value is accessible (default false)). 
+     */
+    void init( unsigned int nb, bool storeValsInStats=false );
+    
+
+    /**
+     * Adds some sample value at the given scale.
+     *
+     * @param[in] idxScale some valid index (according to init).
+     * @param[in] value any value.
+     */
+    void addValue( unsigned int idxScale, float value );
+
+
+    /**
+     * Adds some statistic at the given scale.
+     *
+     * @param idxScale some valid index (according to init).
+     *
+     * @param stat any statistic (which is added to the current
+     * statistic object).
+     */
+    void addStatistic( unsigned int idxScale, const Statistic<float> & stat );
+
+
+
+    /**
+     * It stops and Erase the stats saved values. It must be called to
+     * avoid to store all statistics values when we have to access to
+     * the median value.  Typically if you nedd to access to the
+     * median value of the profile, you need to follow this example:
+     * @code
+     * ScaleProfile sp;
+     * // the value are now stored and you can access to the median value of the profile.
+     * sp.init(true);
+     * sp.addValue(0, 10.5);
+     * sp.addValue(0, 9.2);
+     *  ...
+     * // When all values have been added you can stop to store them again
+     * sp.stopStatsSaving();
+     * // before erasing all statistics data, the median is computed and stored.
+     * @endcode
+     **/
+    void stopStatsSaving() ;    
+
+
+
+
+    // ----------------------- Profile services --------------------------------
+  public:
+    
+    /**
+     * Used to define the method to determine the profile values from
+     * the samples of scale statistics.
+     *
+     * @param type the method applied to the statistics samples: MEAN, MAX, MIN.
+     **/    
+    void setType(ScaleProfileType type);
+    
+
+    
+    /**
+     * Used by boost to Serialize. 
```

Comment:
boost serialization is not used anymore in DGtal, shouldn't this be cleaned up ?